app/services/like_service.rb

```diff
@@ -5,11 +5,17 @@ def initialize(user=nil)
     @user = user
   end
 
-  def create(post_id)
+  def create_for_post(post_id)
     post = post_service.find!(post_id)
     user.like!(post)
   end
 
+  def create_for_comment(comment_id)
+    comment = comment_service.find_by_id!(comment_id)
+    post = post_service.find!(comment.post.id) #checks implicit for public posts
```

Comment:
- Lint/UselessAssignment: Useless assignment to variable - `post`.
- Layout/LeadingCommentSpace: Missing space after `#`.

Reply:
Find a post is used to check if its visible - how can this be better written?